clients/wallet/src/lib.rs

```diff
@@ -3,11 +3,13 @@ use substrate_stellar_sdk::TransactionEnvelope;
 
 pub use horizon::{listen_for_new_transactions, TransactionResponse};
 pub use stellar_wallet::StellarWallet;
+pub use task::*;
 
 pub mod error;
 mod horizon;
 mod stellar_wallet;
+mod task;
 pub mod types;
 
 pub type Ledger = u32;
```

Comment:
I think the lingo is a bit confusing. If I'm not mistaken a `Ledger` is equal to a `Slot`. While 'Ledger' is used to describe a block in the Stellar world, the term 'slot' is rather used in the SCP consensus paper etc. But at the end of the day it boils down to the same thing when it comes to the number we store for a ledger and a slot. For one transaction `x` the slot and ledger will be equal to each other. 

Thus, I would prefer if we find a better way for describing this. If I remember correctly we already have a type for `Slot`, maybe we could just re-use that one. (I know that this is not something you introduced in this PR specifically, it just became a bit more apparent now that we use that term more often.)

Reply:
created a [new issue](https://github.com/pendulum-chain/spacewalk/issues/277) and let's continue from there